src/DataStructures/DataVector.hpp

```diff
@@ -95,6 +96,10 @@ using std::abs;  // NOLINT
  * - step_function: if less than zero returns zero, otherwise returns one
  * - tan
  * - tanh
+ *
+ * `DataVector`s also support a `dot_product` between themselves and with a
+ * `Matrix`, as well as a `magnitude` which is the square root of the dot
```

Comment:
`dot_product` as you've defined it in this PR is for two `DataVectors`, not for a `DataVector` and a `Matrix`